lib/record_store/provider/ns1.rb

```diff
@@ -0,0 +1,222 @@
+          return
+        end
+
+        existing_answers = existing_record["answers"].map { |answer| symbolize_keys(answer) }
+        client.modify_record(
+          zone: zone,
+          fqdn: record_fqdn,
+          type: record.type,
+          params: { answers: existing_answers + new_answers, ttl: record.ttl }
+        )
+      end
+
+      # Deletes an existing record from the zone. It is expected this call modifies external state.
+      #
+      # Arguments:
+      # record - a kind of `Record`
+      def remove(record, zone)
+        record_fqdn = record.fqdn.sub(/.$/, '')
+
+        existing_record = client.record(
+          zone: zone,
+          fqdn: record_fqdn,
+          type: record.type
+        )
+        return if existing_record.nil?
+
+        pruned_answers = existing_record["answers"]
+          .map { |answer| symbolize_keys(answer) }
+          .reject { |answer| answer[:answer] == build_api_answer_from_record(record) }
+
+        if pruned_answers.empty?
+          client.delete_record(
+            zone: zone,
+            fqdn: record_fqdn,
+            type: record.type
+          )
+          return
+        end
+
+        client.modify_record(
+          zone: zone,
+          fqdn: record_fqdn,
+          type: record.type,
+          params: { answers: pruned_answers }
+        )
+      end
+
+      # Updates an existing record in the zone. It is expected this call modifies external state.
+      #
+      # Arguments:
+      # id - provider specific ID of record to update
+      # record - a kind of `Record` which the record with `id` should be updated to
+      def update(id, record, zone)
+        record_fqdn = record.fqdn.sub(/.$/, '')
+
+        existing_record = client.record(
+          zone: zone,
+          fqdn: record_fqdn,
+          type: record.type,
+          must_exist: true,
+        )
+
+        # Identify the answer in this record with the matching ID, and update it
+        updated = false
+        existing_record["answers"].each do |answer|
+          if answer["id"] == id
+            updated = true
+            answer["answer"] = build_api_answer_from_record(record)
+          end
+        end
+
+        raise(Error, "while trying to update a record, could not find answer with fqdn: #{record.fqdn}, type; #{record.type}, id: #{id}") unless updated
+
+        client.modify_record(
+          zone: zone,
+          fqdn: record_fqdn,
+          type: record.type,
+          params: { answers: existing_record["answers"], ttl: record.ttl }
+        )
+      end
+
+      def build_from_api(api_record, zone)
+        fqdn = Record.ensure_ends_with_dot(api_record["domain"])
+
+        record_type = api_record["type"]
+        api_record["answers"].map do |api_answer|
+          answer = api_answer["answer"]
+          record = {
+            ttl: api_record["ttl"],
+            fqdn: fqdn.downcase,
+            record_id: api_answer["id"]
+          }
+          return if record_type == 'SOA'
```

Comment:
If you want to exit because of the record type, why doing it here in the `map` instead of just after reading the `record_type` value before the loop ?